matchms/similarity/CosineGreedyVectorial.py

```diff
@@ -0,0 +1,123 @@
+from typing import Tuple
+import numpy
+from matchms.typing import SpectrumType
+
+
+class CosineGreedyVectorial:
```

Comment:
Why add `Vectorial` to name? It uses some vector based numpy methods, but also a for loop.

Reply:
I picked "vectorial" because it was built around numpy arrays to avoid further loops. But that's not set in stone, so would you have a better suggestion?

Reply:
I don't have a better name